### PR TITLE
implement mini-epochs in training

### PR DESCRIPTION
**Pull request by ftostevin-ont**
This PR implements mini-epochs (multiple rounds of validation per full traversal of the training data) in Train.py.
I have found that running with `--mini_epochs 5` slightly improves the accuracy of the resulting model, since models at intermediate stages of an epoch may be better than those at the end of an epoch. If no `--mini_epochs` parameter is provided, training should be the same as currently.

To facilitate this change, training batch generation has been reimplemented as a `keras.utils.Sequence` class, since this has an `on_epoch_end` method that can be used to count miniepochs and only shuffle when a full epoch has been completed. This code is based on the `SequenceBatcher` [implemented in Medaka](https://github.com/nanoporetech/medaka/blob/cd07dc4fa62954cbc95990fedc04c6b19371d2cf/medaka/keras_ext.py#L67).

### CODE REVIEW

**Comment by aquaskyline** [comment]
Testing with fine-tuning a model. Will roll out with r7.
